### PR TITLE
Fix first doc comment inside macro yield

### PR DESCRIPTION
**Pull request by RX14**
The following example eats the doc comment:

```cr
macro foo
  {{yield}}
end

foo do
  # doc comment
  def test
  end
end
```

This is because the first line of comment is generated on the same line as the `begin` which is inserted when using `{{yield}}`, like so:

```cr
begin # doc comment
  def test
  end
end
```

Using a newline instead of whitespace after `begin` in macro yield fixes this.

### CODE REVIEW

File: src/compiler/crystal/macros/interpreter.cr [hunk 1]
```diff
@@ -104,7 +104,7 @@ module Crystal
         if (loc = @last.location) && loc.filename.is_a?(String) || is_yield
           macro_expansion_pragmas = @macro_expansion_pragmas ||= {} of Int32 => Array(Lexer::LocPragma)
           (macro_expansion_pragmas[@str.pos.to_i32] ||= [] of Lexer::LocPragma) << Lexer::LocPushPragma.new
-          @str << "begin " if is_yield
+          @str << "begin\n" if is_yield
           @last.to_s(@str, macro_expansion_pragmas: macro_expansion_pragmas, emit_doc: true)
           @str << " end" if is_yield
```
Comment (Sija) [inline src/compiler/crystal/macros/interpreter.cr:109]: Perhaps this should be changed accordingly?

Comment (straight-shoota) [reply src/compiler/crystal/macros/interpreter.cr:109]: I don't think there's a situation where this could cause harm.

**Comment by straight-shoota** [comment]
I added a spec for this using the example from the OP.

**Review by straight-shoota — APPROVED** [review]

File: spec/compiler/semantic/macro_spec.cr [hunk 2]
```diff
@@ -613,6 +613,21 @@ describe "Semantic: macro" do
       CRYSTAL
   end
 
+  it "being with {{ yield }} (#15050)" do
```
Comment (ysbaddaden) [inline spec/compiler/semantic/macro_spec.cr:616]: Did you mean?
```suggestion
  it "begins with {{ yield }} (#15050)" do
```

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**